main.cc

```diff
@@ -932,6 +932,11 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 tm_cfg.topo_cfg.disable_proximity_sorting = true;
             }
             token_metadata.start([] () noexcept { return db::schema_tables::hold_merge_lock(); }, tm_cfg).get();
+            token_metadata.invoke_on_all([&] (auto& stm) {
+                stm.set_stall_detector_threshold(
+                        std::chrono::duration_cast<std::chrono::steady_clock::duration>(
+                                std::chrono::duration<double>(cfg->topology_barrier_stall_detector_threshold_seconds())));
+            }).get();
```

Comment:
Why not add the parameter to tm_cfg?

Reply:
I thought about it, but considered it unkosher because tm_cfg configures topology, but stall detection is part of stm machinery.

Reply:
What about passing it through the constructor then?